pkg/cache/cache_from.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"fmt"
+
+	"github.com/okteto/okteto/pkg/constants"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+)
+
+// CacheFrom is a list of images to import cache from.
+type CacheFrom []string
+
+type ImageCtrlInterface interface {
+	GetRegistryAndRepo(image string) (string, string)
+	GetRepoNameAndTag(repo string) (string, string)
+}
+
+type oktetoRegistryInterface interface {
+	GetImageCtrl() ImageCtrlInterface
+	HasGlobalPushAccess() (bool, error)
+}
+
+// UnmarshalYAML implements the Unmarshaler interface of the yaml pkg.
+func (cf *CacheFrom) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var single string
+	err := unmarshal(&single)
+	if err == nil {
+		*cf = CacheFrom{single}
+		return nil
+	}
+
+	var multi []string
+	err = unmarshal(&multi)
+	if err == nil {
+		*cf = multi
+		return nil
+	}
+
+	return err
+}
+
+// MarshalYAML implements the marshaler interface of the yaml pkg.
+func (cf CacheFrom) MarshalYAML() (interface{}, error) {
+	if len(cf) == 1 {
+		return cf[0], nil
+	}
+
+	return cf, nil
+}
+
+// AddDefaultPullCache appends the default cache layers for a given image
+func (cf *CacheFrom) AddDefaultPullCache(reg oktetoRegistryInterface, image string) {
```

Comment:
It feels odd that an exported function depends on an unexported interface. The compiler will catch any possible error anyways but it is hard for a client of this package to know which functions has to implement to satisfy the interface if it is unexported unless they access to the source code. 

Imagine if someone is using the CLI as dependency in their project and they want to use the CacheFrom. They would have to go to the source code to see which functions are defined in that interface